jib-core/src/main/java/com/google/cloud/tools/jib/image/json/V22ManifestListTemplate.java

```diff
@@ -72,6 +75,16 @@
   private final int schemaVersion = SCHEMA_VERSION;
   private final String mediaType = MANIFEST_MEDIA_TYPE;
 
+  @JsonCreator
+  public V22ManifestListTemplate(@JsonProperty("mediaType") String mediaType, @JsonProperty("schemaVersion") Integer schemaVersion, @JsonProperty("manifests") List<ManifestDescriptorTemplate> manifests) {
+    Preconditions.checkArgument(Objects.equals(mediaType, MANIFEST_MEDIA_TYPE), "mediaType " + mediaType + " does not equal " + MANIFEST_MEDIA_TYPE);
+    Preconditions.checkArgument(Objects.equals(schemaVersion, SCHEMA_VERSION), "schemaVersion " + schemaVersion + " does not equal " + SCHEMA_VERSION);
+    this.manifests = manifests;
+  }
```

Comment:
I guess the behavior is different in that, previously, when deserializing a JSON that stores a value different from `SCHEMA_VERSION`, the code does not fail and the Java object can always force the value `SCHEMA_VERSION` defined in this Java code. OTOH, I think the new code will crash at runtime. (However, I admit that it's extremely unlikely that a serialized JSON has different constants than what we expect.)

Reply:
I'm not super in love with this either, we could always just log if the mediaType or schemaVersion aren't what we expect.